src/coreclr/jit/llvm.cpp

```diff
@@ -952,6 +976,20 @@ Value* Llvm::consumeValue(GenTree* node, Type* targetLlvmType)
                     trueNodeType = TYP_UBYTE;
                     break;
 
+                case GT_CAST:
+                    switch (nodeValue->getType()->getPrimitiveSizeInBits())
```

Comment:
Yep, for casts the `gtCastType` is the "true node type", so this can be just `trueNodeType = node->AsCast()->CastToType();`